build.gradle.kts

```diff
@@ -0,0 +1,32 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {
+    repositories {
+        google()
+        jcenter()
+    }
+    dependencies {
+        classpath("com.android.tools.build:gradle:3.4.1")
+        // Due an issue the full qualified name is required here. See https://github.com/gradle/kotlin-dsl/issues/1291
+        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${to.dev.dev_android.build.BuildConfig.kotlinVersion}")
```

Comment:
My Gradle sync failed with `Unresolved reference: build` and it's pointing to the fully qualified name.

```
Script compilation errors:

  Line 11:         classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${to.dev.dev_android.build.BuildConfig.kotlinVersion}")
                                                                          ^ Function invocation 'to(...)' expected

  Line 11:         classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${to.dev.dev_android.build.BuildConfig.kotlinVersion}")
                                                                          ^ No value passed for parameter 'that'

  Line 11:         classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${to.dev.dev_android.build.BuildConfig.kotlinVersion}")
                                                                             ^ Unresolved reference: dev

3 errors
```
Did you had to do anything extra to get this to work?

Reply:
No, normally not.  As mention by @sierisimo the buildSrc ist compiled and added to the classpath of every Gradle build script.

You synced it in Android Studio? It worked for me, with Android Studio 3.4 out of the box and without any cleanup. Nevertheless, you could try to execute the clean task once just to be sure that Gradle (and/or your IDE) didn't cache strange stuff and builds the buildSrc. Sometimes AS has a strange behavior when changing larger parts of build scripts (also with Groovy scripts). Reimport of the project, also once just to AS fix itself, might help as well.

Reply:
Yeah I synced it, invalidated cache & restart a few times, reimport the project and it still isn't working for me.

Reply:
Android Studio is not 100% to fully support Kotlin DSL (even IntelliJ has it still in an early support). So my steps to check this works is:

1. Check your `gradle/wrapper/gradle-wrapper.properties` is pointing to `distributionUrl=https\://services.gradle.org/distributions/gradle-5.1.1-all.zip`
2. Run the gradle task from the terminal `./gradlew assembleDebug` or something like it. Probably `./gradlew build` will give you the more specific error.

If after checking this two the error still happens is probably a bad idea to go with the Kotlin DSL as it won't work for all developers of the app

Reply:
Worth to say: If your `gradle/wrapper/gradle-wrapper.properties` is not pointing to the right version of gradle, that's a required change in this PR

Reply:
@maestromac Can I help you to solve the problem with the sync? Any update about it?

Reply:
Sorry for the late reply here. @sierisimo your step one points to gradle-5.1.1 while this PR updated it to 5.4.1. Mine is already pointed to 5.4.1 and it doesn't work. I also can't get it to work on my windows machine, same error. Output [here](https://gist.github.com/maestromac/9165abd839ff2f85700bdbc0d7776d1e)

Reply:
Oh, I've failed hard. Due the gitignore of the project the buildSrc/src folder was ignored. So I didn't commit the Config.kt and unfortunately oversee it after your comments. I changed the gitignore and commited the file.

Sorry for the inconvenience and a giphy for all the trouble.

![Failed](https://media.giphy.com/media/NQL7Wuo2JSQSY/giphy.gif)

Reply:
That's alright 😄 . It's working for me now.